tests/tests/messages.rs

```diff
@@ -186,6 +191,105 @@ async fn messages_empty_results_for_owner_with_no_messages(
     assert_eq!(result.results.len(), 0);
 }
 
+#[tokio::test]
+async fn message_status__can_get_unspent() {
+    // Given
+    let owner = Address::new([1; 32]);
+    let nonce = 1.into();
+    let amount = 1_000;
+
+    let msg = MessageConfig {
+        recipient: owner,
+        nonce: 1.into(),
+        amount,
+        ..Default::default()
+    };
+
+    let mut config = Config::local_node();
+    config.chain_conf.initial_state = Some(StateConfig {
+        messages: Some(vec![msg]),
+        ..Default::default()
+    });
+
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    // When
+    let status = client.message_status(&nonce).await.unwrap();
+
+    // Then
+    assert_eq!(status, MessageStatus::Unspent);
+}
+
+#[tokio::test]
+async fn message_status__can_get_spent() {
+    // Given
+    let alice = Address::from([1; 32]);
+    let bob = Address::from([2; 32]);
+    let charlie = Address::from([3; 32]);
+
+    let owner = Address::new([1; 32]);
+    let nonce = 1.into();
+    let amount = 1_000;
+
+    let msg = MessageConfig {
+        recipient: owner,
+        nonce: 1.into(),
+        amount,
+        ..Default::default()
+    };
+
+    let mut config = Config::local_node();
+    config.chain_conf.initial_state = Some(StateConfig {
+        messages: Some(vec![msg]),
+        ..Default::default()
+    });
+
+    let srv = FuelService::new_node(config).await.unwrap();
+    let client = FuelClient::from(srv.bound_address);
+
+    let input =
+        Input::message_coin_signed(charlie, alice, amount, nonce, Default::default());
```

Comment:
It is better to use naming `sender` and `recipient` instead. Plus `alice` and `owner` share the same address=)

Reply:
Whoops. I know why that happened. I'll clean it up.